### ISSUE TITLE
dotnet run fails if run from a directory named `;)`

### ISSUE THREAD

**Issue by martincostello** [dotnet/sdk#13954]
_Found by [someone on StackOverflow](https://stackoverflow.com/a/64168215/1064169)._

If `dotnet run` is executed from a folder named `;)` then the SDK fails with the following error:

```
Failed to create CoreCLR, HRESULT: 0x80070057
```

## Steps to reproduce

```powershell
> mkdir ";)"
> cd '.\;)\'
> dotnet new console
> dotnet run
```

## Expected Result

```
Hello world!
```

## Actual Result

```
Failed to create CoreCLR, HRESULT: 0x80070057
```

## Runtime information

```
> dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.1.20452.10
 Commit:    473d1b592e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.1.20452.10\

Host (useful for support):
  Version: 5.0.0-rc.1.20451.14
  Commit:  38017c3935

.NET SDKs installed:
  2.1.809 [C:\Program Files\dotnet\sdk]
  3.1.104 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.303 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.8.20417.9 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.1.20452.10 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-preview.8.20414.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.1.20451.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-preview.8.20407.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20451.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-preview.8.20411.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.1.20452.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by albertomandlate** [comment]
I'm having the same problem, and I don't have any folder containing any of those characters.

**Comment by sp1ritCS** [comment]
I've tried to run the [build-dotnet-tarball](https://src.fedoraproject.org/rpms/dotnet5.0/raw/rawhide/f/build-dotnet-tarball) script in a directory containing `:` (colon). It failed with the same error. (Linux, openSUSE Tumbleweed)

**Comment by KRoy118** [comment]
Yeah.I tried doing the same.Saved a file with ";" but didn't work.IT replaced the nampespace with '_"

**Comment by princefishthrower** [comment]
For what it's worth, also hit this after running Nswag with an API generation script that's been otherwise working for the past year. I suspect it also has to do with the fact I installed the .NET 6 SDK recently.

**Comment by Evangelink** [comment]
It seems to be linked to the `;` character as a folder named `;` reproduce the issue. 

From my tests `dotnet restore` and `dotnet build` are fine but not `dotnet run`. I believe this is another case of encoding of special char that was forgotten. I will check if #23730 might help with this issue.

**Comment by baronfel** [comment]
I think we've gotten a bit better since this was last touched on - on .NET SDK 6.0.301 I can run the repro steps from the first comment on a Debian WSL instance and the newly-generated application runs.

Can folks reproduce the error on 6.0.301? If not, I'll close this.

**Comment by martincostello** [comment]
This still repros for me on Windows 11 with the 6.0.301 (and .NET 7 preview 5) SDKs.

**Comment by baronfel** [comment]
Thanks for confirming @martincostello, let me try again on windows. I mistakenly thought your initial repro was linux despite you providing your `dotnet --info` output.

**Comment by danbopes** [comment]
We got this on windows, after publishing an application to a windows box. (.net 9-windows)
